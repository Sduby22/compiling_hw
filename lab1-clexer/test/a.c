//test
/* test
 * test
 */
#define TEST

long foo(short a) {
	return a + 0x1f2b3c;
}

int main() {
	const float asd = 123.20312;
	static char *assdsad = "asdsad\
			       asdsadsad";
	char ch = '\n';
	if (1) {} else {}
	switch(1) {case '1': default:;}
	for (;;) {break;}
	while(0) {continue;}
	int a = sizeof(double);
	struct asd;
	union c;
	return 0;
}
