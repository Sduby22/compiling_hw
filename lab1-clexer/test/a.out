SYMTYPE	SYMATTR	SYMVALUE
macro
T_LONG	---	long
T_ID	---	foo
(	---	(
T_SHORT	---	short
T_ID	---	a
)	---	)
{	---	{
T_RETURN	---	return
T_ID	---	a
+	---	+
T_NUM	ATTR_NUM_HEX	0x1f2b3c
...
T_NUM	ATTR_NUM_FLOAT	123.20312
...
T_LITERAL	---	"asdsad\
			       sadsad\
			       asdsadsad\
			       asdsadsad"
;	---	;
T_CHAR	---	char
...
T_UNION	---	union
T_ID	---	c
;	---	;
T_RETURN	---	return
T_NUM	ATTR_NUM_DECIMAL	0
;	---	;
}	---	}
代码行数 : 28
总字节数 : 396
注释数 : 2
关键字数 : 24
标点符号数 : 49
数字常量数 : 8
字符串常量数 : 1
标识符数 : 10
未识别符号数 : 0
